mmf/utils/distributed.py

```diff
@@ -66,7 +77,14 @@ def broadcast_tensor(tensor, src=0):
         return tensor
 
     with torch.no_grad():
-        dist.broadcast(tensor, src=0)
+        if is_xla():
+            tensor = xm.all_to_all(
+                    tensor.repeat([world_size,1]),
+                    split_dimension=0, 
+                    concat_dimension=0, 
+                    split_count=world_size)[0] 
```

Comment:
deindent:

```python
            tensor = xm.all_to_all(
                tensor.repeat([world_size,1]),
                split_dimension=0, 
                concat_dimension=0, 
                split_count=world_size,
            )[0] 

```

Reply:
oh actually, `all_to_all` is not broadcast? We'd need to be more careful here. we don't have a corresponding function.

Reply:
where do they use broadcast? does the current model use this? if so, it won't work i think.